tools/kicad/kicad_functions/src/vdiv.rs

```diff
@@ -0,0 +1,128 @@
+    let calc = RCalc::new(vec![config.series; config.count]);
+
+    calc.calc(|set| {
+        if let Some(true) = config.resistance_min.map(|r| set.sum() < r) {
+            return None; // Sum of resistance less than minimum
+        }
+
+        if let Some(true) = config.resistance_max.map(|r| set.sum() > r) {
+            return None; // Sum of resistance larger than maximum
+        }
+
+        // TODO: Storing this externally and using interior mutability
+        //  could be helpful to avoid reallocating on every invocation.
+        let mut context = HashMapContext::new();
+        for i in 1..=config.count {
+            context
+                .set_value(format!("R{}", i).into(), Value::Float(set.r(i)))
+                .unwrap();
+        }
+
+        match config.expression.eval_with_context(&context) {
+            Ok(v) => Some((config.target - v.as_number().unwrap()).abs()),
+            Err(e) => match &e {
+                EvalexprError::DivisionError { divisor: d, .. } => {
+                    if let Ok(n) = d.as_number() {
+                        if n == 0.0 {
+                            // This soft-catch may be a bit redundant. Based on some testing the
+                            // internal conversions in evalexpr cause zero values to deviate
+                            // slightly from zero, thus avoiding division by zero even if you
+                            // explicitly write a zero division into the voltage divider equation.
+                            return None;
+                        }
+                    }
+                    panic_any(e) // No graceful way to handle this from the closure
+                }
+                _ => panic_any(e),
+            },
+        }
+    })
+}
+
+/// `voltage_divider` computes values for resistor-based voltage dividers.
+/// - Usage: vdiv(<target voltage>, <divider expression>, <resistor series>, (min resistance), (max resistance))
+/// - Example: vdiv(5.1, "(R1+R2)/R2*0.8", "E96", 500e3, 700e3)
+/// - Output: (<closest voltage>, <R1 value>, <R2 value>, ...)
+/// There can be arbitrary many resistors in the divider, but they must be named "R1", "R2", etc.
+/// The computed optimal resistance values are also presented in this order. The minimal and maximal
+/// resistance limits are optional parameters, and only consider the sum of the resistances of all
+/// resistors defined in the expression.
+pub(crate) fn voltage_divider(argument: &Value) -> EvalexprResult<Value> {
+    let config = VoltageDividerConfig::parse(argument)?;
+    if let Some(res) = calculate(&config) {
+        // Take the first result, these are ordered by increasing error
+        if let Some((v, set)) = res.iter().next() {
+            let voltage = config.target + *v as f64 / 1e9;
+            let mut tuple = vec![Value::from(voltage)];
+            for i in 1..=config.count {
+                tuple.push(Value::from(set.r(i)));
+            }
+            return Ok(Value::from(tuple));
```

Comment:
Does evalexpr support tuples natively, or are they arrays?

Reply:
The "tuple" is just a `Vec` with any amount of elements. `evalexpr` has a an enum variant `Value(TupleType)`, where `TupleType` is an alias for `Vec<Value>`. I haven't checked what is supported for those though, i.e. indexing is pretty much required for this use case. Will check in a bit if that's implemented.

Reply:
Ok, thanks for the info